src/launcher/index.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import {
+  Token
+} from 'phosphor/lib/core/token';
+
+import {
+  JSONObject
+} from 'phosphor/lib/algorithm/json';
+
+import {
+  DisposableDelegate, IDisposable
+} from 'phosphor/lib/core/disposable';
+
+
+/* tslint:disable */
+/**
+ * The main menu token.
+ */
+export const ILauncher = new Token<ILauncher>('jupyter.services.launcher');
+/* tslint:enable */
+
+
+/**
+ * The launcher interface.
+ */
+export interface ILauncher {
+  /**
+   * Add a command item to the Launcher
+   *
+   * @param name - The display name.
+   *
+   * @param action - The command that should be executed on clicking.
+   *
+   * @param args - Arguments to the `action` command.
+   *
+   * @param imgName - The CSS class to attach to the item. Defaults to
```

Comment:
By our CSS class naming standards, I think we should have two classes: `jp-Launcher-item` and `jp-Launcher-thisCommand`. That way it is possible to style all items and some items.